packages/block-editor/src/components/block-navigation/block.js

```diff
@@ -63,38 +62,62 @@ export default function BlockNavigationBlock( {
 		[ clientId ]
 	);
 
-	const { selectBlock: selectEditorBlock } = useDispatch( blockEditorStore );
+	const {
+		selectBlock: selectEditorBlock,
+		toggleBlockHighlight,
+	} = useDispatch( blockEditorStore );
 
 	const hasSiblings = siblingBlockCount > 0;
 	const hasRenderedMovers = showBlockMovers && hasSiblings;
-	const hasVisibleMovers = isHovered || isFocused;
 	const moverCellClassName = classnames(
 		'block-editor-block-navigation-block__mover-cell',
-		{ 'is-visible': hasVisibleMovers }
+		{ 'is-visible': isHovered }
 	);
 	const {
 		__experimentalFeatures: withExperimentalFeatures,
+		__experimentalPersistentListViewFeatures: withExperimentalPersistentListViewFeatures,
 	} = useBlockNavigationContext();
 	const blockNavigationBlockSettingsClassName = classnames(
 		'block-editor-block-navigation-block__menu-cell',
-		{ 'is-visible': hasVisibleMovers }
+		{ 'is-visible': isHovered }
 	);
 	useEffect( () => {
-		if ( withExperimentalFeatures && isSelected ) {
+		if (
+			( withExperimentalFeatures ||
+				withExperimentalPersistentListViewFeatures ) &&
+			isSelected
+		) {
 			cellRef.current.focus();
 		}
-	}, [ withExperimentalFeatures, isSelected ] );
+	}, [
+		withExperimentalFeatures,
+		withExperimentalPersistentListViewFeatures,
+		isSelected,
```

Comment:
This dep list is why trying `await clearSelectedBlock` before `selectBlock` was not solving issue 3.  If I make this an empty array and then try that solution, it starts working!  

I guess since `isSelected` will toggle from true -> false -> true in that situation, this `.focus` fires as the last event and focus remains in sidebar as opposed to moving to that selected block.  I admit that I have no idea why this isn't an issue normally, since it behaves that way I would expect this to draw focus back to the sidebar anytime I try to have it select a block (whether it is already selected or not). Either way, it definitely seems to be what is stopping the clear/reselect approach for working as expected.

I don't think there is any reason for this dep array not to be blank?  As far as I can tell the only time we would want to focus an item in the sidebar would be on mount.  But having `isSelected` as a dependency means `.focus()` will be called on this sidebar item when its corresponding block becomes the selected block, which seems quite problematic.